Dockerfile

```diff
@@ -18,12 +18,24 @@ RUN cargo build --release --bin sinker
 # We do not need the Rust toolchain to run the binary!
 FROM debian:bookworm-slim@sha256:f80c45482c8d147da87613cb6878a7238b8642bcc24fc11bad78c7bec726f340
 
+# Update the system and install necessary packages
 RUN apt update \
     && apt install --yes ca-certificates libssl3 --no-install-recommends \
-    && rm -rf /var/lib/{apt,dpkg,cache,log} \
-    && groupadd --gid 1500 sinker \
+    && rm -rf /var/lib/{apt,dpkg,cache,log}
+
+# Remove setuid/setgid bits from executables. Also If the path is /proc, don't descend into it. This effectively excludes /proc from the search.
+RUN find / -path /proc -prune -o -type f \( -perm -4000 -o -perm -2000 \) -exec chmod a-s {} \;
+
+# Create a dedicated user and group for the application
+RUN groupadd --gid 1500 sinker \
     && useradd --uid 1500 --gid sinker --shell /bin/bash --create-home sinker
 
+# RUN apt update \
+#     && apt install --yes ca-certificates libssl3 --no-install-recommends \
+#     && rm -rf /var/lib/{apt,dpkg,cache,log} \
+#     && groupadd --gid 1500 sinker \
+#     && useradd --uid 1500 --gid sinker --shell /bin/bash --create-home sinker
```

Comment:
Can you elaborate a little bit on why these changes are necessary?

Reply:
> That said, the addition of -path /proc -prune causes nothing to match.

CORRECTION: I did something wrong when I tested this. I can see now that `find / -path /proc -prune -o -type f \( -perm -4000 -o -perm -2000 \) -ls` does match stuff. This might be a refinement you'd still be interested in though:
```
# Remove setuid/setgid bits from executables as a hardening measure so non-root processes can't escalate.
RUN find / \( -path /dev -o -path /proc -o -path /sys \) -prune -o -type f \( -perm -4000 -o -perm -2000 \) -exec chmod a-s {} \;
```